### PR TITLE
[Bugfix][Metrics][MFU] Size activation traffic from the model dtype

### PR DESCRIPTION
**Pull request by thillai-c**
## Purpose

`BaseConfigParser` sizes weight traffic from the model's dtype but pins activation traffic to 2 bytes:

```python
args.weight_byte_size = get_dtype_size(torch_dtype)

# FIXME: handle this better by parsing whether activations use
# bf16, fp32, etc...
args.activation_byte_size = 2
```

`torch_dtype` is already resolved on the line above, so the two values come from the same config and disagree. For a model served with `--dtype float32`, every activation term in the MBU accounting is counted at half its real size while that same model's weight terms are correct. Activations are produced in the model's compute dtype, so this keys them on it.

`activation_byte_size` is set once here and never overridden. Quantization parsers override `weight_byte_size` (fp8, mxfp4 and friends) but leave activations alone, which is why the model dtype remains the right source after this change.

## Impact

- bfloat16 and float16: unchanged, 2 bytes, which covers the overwhelming majority of deployments.
- float32: 2 to 4 bytes. Activation read and write traffic doubles and reported MBU drops accordingly.

The float32 change is user visible. Anyone tracking MBU for a float32 deployment will see the number move; that movement is the correction, since the previous value understated memory traffic.

## Test Plan

```bash
pytest -v tests/v1/metrics/test_perf_metrics.py
pre-commit run ruff-check ruff-format --files \
  vllm/v1/metrics/perf.py tests/v1/metrics/test_perf_metrics.py
```

New tests in the existing file:

- `test_base_config_parser_activation_byte_size_follows_dtype`, parametrized over float16, bfloat16 and float32, asserting activation and weight byte sizes agree with the dtype.
- `test_attention_read_bytes_scale_with_activation_dtype`, which goes through `AttentionMetrics.get_read_bytes_breakdown` and asserts `qkv_input`, a pure activation term, is exactly twice as large for float32 as for bfloat16.

The existing `test_base_config_parser` assertion of `activation_byte_size == 2` for float16 still holds and is unchanged; only its comment was updated, since 2 is now the dtype's size rather than a default.

## Test Result

```text
72 passed, 1 warning in 19.45s
```

That is the 68 pre-existing tests plus the 4 added here.

Linting:

```text
ruff check...............................................................Passed
ruff format..............................................................Passed
```

No model evaluation: this changes analytic MFU/MBU accounting only. No kernel, no model computation, and no output logits are affected.

## AI assistance

AI assistance (Claude Code) was used to locate this and draft the tests. I reviewed every changed line and ran the commands above myself.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by thillai-c** [comment]
@markmc  mind taking a look? float32 models under-report activation traffic because activation_byte_size was hardcoded to 2. Resolves the FIXME at perf.py:383.

**Review by markmc — APPROVED** [review]
lgtm, thanks
